---
title: C6320
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: bd2cc95cf0dd14f4ee429cf542a54fcb53947ea4
ms.sourcegitcommit: 5f6ad1cefbcd3d531ce587ad30e684684f4c4d44
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 10/22/2019
ms.locfileid: "72746327"
---
# <a name="c6320"></a>C6320
предупреждение C6320: выражение фильтра исключений является константой EXCEPTION_EXECUTE_HANDLER. Это может маскировать исключения, которые не предназначены для обработки

 Это предупреждение указывает на побочный результат использования константы EXCEPTION_EXECUTE_HANDLER в блоке __except. В этом случае инструкция в блоке \__except всегда будет выполняться для устранения исключения, включая исключения, которые вы не хотите обработать в определенной функции. Рекомендуется проверять исключение перед его обработкой.

## <a name="example"></a>Пример
 В следующем коде создается это предупреждение, поскольку блок __except попытается выполнить обработку исключений всех типов:

```
#include <stdio.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;  // might cause access violation exception
      // code ...
   }
   __except(EXCEPTION_EXECUTE_HANDLER) // warning
   {
      puts("in except");
      // code ...
   }
}
```

 Чтобы устранить это предупреждение, используйте `GetExceptionCode` для проверки конкретного исключения перед его обработкой, как показано в следующем коде:

```
#include <stdio.h>
#include <windows.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;    // might cause access violation exception
      // code ...
   }
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
   {
      puts("in except");
      // code ...
   }
}
```

## <a name="see-also"></a>См. также
 [Оператор try-except](/cpp/cpp/try-except-statement)
